ansible_events_ui/api/project.py

```diff
@@ -0,0 +1,125 @@
+import sqlalchemy as sa
+from fastapi import APIRouter, Depends, HTTPException
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ansible_events_ui import schemas
+from ansible_events_ui.db import models
+from ansible_events_ui.db.dependency import get_db_session
+from ansible_events_ui.project import clone_project, sync_project
+
+router = APIRouter()
+
+
+@router.get("/api/projects/")
+async def list_projects(db: AsyncSession = Depends(get_db_session)):
+    query = sa.select(
+        models.projects.c.id, models.projects.c.url, models.projects.c.name
+    )
+    result = await db.execute(query)
+    return result.all()
+
+
+@router.post("/api/projects/")
+async def create_project(
+    project: schemas.ProjectCreate, db: AsyncSession = Depends(get_db_session)
+):
+    found_hash, tempdir = await clone_project(project.url, project.git_hash)
+    project.git_hash = found_hash
+    query = sa.insert(models.projects).values(
+        url=project.url, git_hash=project.git_hash, name=project.name, description=project.description
+    )
+    try:
+        result = await db.execute(query)
+    except sa.exc.IntegrityError:
+        raise HTTPException(status_code=422, detail="Unprocessable Entity.")
+
+    (project_id,) = result.inserted_primary_key
+    await sync_project(project_id, tempdir, db)
+    await db.commit()
+    return {**project.dict(), "id": project_id}
+
+
+@router.get("/api/projects/{project_id}", response_model=schemas.ProjectRead)
```

Comment:
Could you add the `operation_id` param for the openapi spec?
```suggestion
@router.get("/api/projects/{project_id}", response_model=schemas.ProjectRead, operation_id="show_project")
```

Reply:
shouldn't the rest of the endpoints also have `response_model` defined?